tests/integrationv2/utils.py

```diff
@@ -14,7 +16,7 @@ def get_expected_s2n_version(protocol, provider):
     tls12 is always chosen. This is true even when the requested
     protocol is less than tls12.
     """
-    if provider == S2N and protocol != Protocols.TLS13:
+    if (provider == S2N or issubclass(provider, S2N)) and protocol != Protocols.TLS13:
```

Comment:
nit: the `provider == S2N` check should be unnecessary with the `issubclass(provider, S2N)` check.

```
>>> class A:
...     pass
... 
>>> issubclass(A, A)
True
```